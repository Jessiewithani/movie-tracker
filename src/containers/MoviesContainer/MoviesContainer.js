import React from 'react';
import Card from '../Card/Card';
import PropTypes from 'prop-types';
import images from '../../assets/images';
import '../MoviesContainer/MoviesContainer.scss'
import { connect } from 'react-redux'

export const MoviesContainer = ({ movies }) => {
  const movieCards = movies.map(movie => {
      return (
        <Card 
        key={movie.id}
        poster={`https://image.tmdb.org/t/p/w342${movie.poster_path}`}
        title={movie.title}
        favorited="false"
        />
        )
      })

  return (
    <div className="scroll-wrapper">
      <div className="movies-container">
        <img className="left-arrow arrow" src={images.leftArrow} alt="left arrow"/>
        {movieCards}
        <img className="right-arrow arrow" src={images.rightArrow} alt="right arrow"/>
      </div>
    </div>
  )
}

const mapStateToProps = ({ movies }) => ({
  movies
})

export default connect(mapStateToProps, null)(MoviesContainer)
