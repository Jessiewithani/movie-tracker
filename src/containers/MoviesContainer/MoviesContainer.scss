.scroll-wrapper {
    width: 100vh;
    height: 100vw;
    transform: rotate(-90deg) translateX(-100vh);
    transform-origin: top left;
    overflow-y: scroll;
    overflow-x: hidden;
    position: absolute;
}

.movies-container {
    background-color: teal;
    display:flex;
    justify-content: center;
    flex-direction: row;
    width: fit-content;
    transform: rotate(90deg) translateY(-105vh);
    transform-origin: top left;
    // padding: 0 8%;
}

.arrow {
    height: 100px;
    width: auto;
    margin: auto;
}

