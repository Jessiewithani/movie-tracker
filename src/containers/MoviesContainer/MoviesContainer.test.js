import { MoviesContainer, mapStateToProps }from './MoviesContainer';
import React from 'react';
import { shallow } from 'enzyme';

describe('MoviesContainer', () => {
  let wrapper;
  beforeEach(() => {
    let mockMovies = [
      {
        title: "Joker",
        id: 4,
        release_date: "2019-10-18"
      },
      {
        title: "Maleficent: Mistress of Evil",
        id: 56,
        release_date: "2019-10-04"
      },
      {
        title: "Zombieland: Double Tap",
        id: 70,
        release_date: "2019-10-18"
      }
    ]
    let mockFavorites = [{
      title: "Maleficent: Mistress of Evil",
      id: 56,
      release_date: "2019-10-04"
    },
    {
      title: "Zombieland: Double Tap",
      id: 70,
      release_date: "2019-10-18"
    }]
    wrapper = shallow(
      <MoviesContainer
      toggleFavorites={jest.fn()}
      movies={mockMovies}
      favorites={mockFavorites}
      />
    )
  })

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot()
  })
})

describe('mapStateToProps in MoviesContainer', () => {
  it('should return an array of movies', () => {
    const mockState = {
      movies: [{
        title: "Joker",
        id: 4,
        release_date: "2019-10-18"
      },
      {
        title: "Maleficent: Mistress of Evil",
        id: 56,
        release_date: "2019-10-04"
      },
      {
        title: "Zombieland: Double Tap",
        id: 70,
        release_date: "2019-10-18"
      }],
      saveMovies: 'SAVE_MOVIES'
    }

    const expected = {
      movies: [{
        title: "Joker",
        id: 4,
        release_date: "2019-10-18"
      },
      {
        title: "Maleficent: Mistress of Evil",
        id: 56,
        release_date: "2019-10-04"
      },
      {
        title: "Zombieland: Double Tap",
        id: 70,
        release_date: "2019-10-18"
      }]
    }

    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected)
  })
})