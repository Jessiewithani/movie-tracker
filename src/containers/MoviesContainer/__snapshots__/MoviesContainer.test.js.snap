// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesContainer should match the snapshot with all data passed in 1`] = `
<div
  className="scroll-wrapper"
>
  <div
    className="movies-container"
  >
    <img
      alt="left arrow"
      className="left-arrow arrow"
      src="right-arrow.png"
    />
    <Card
      favorited={true}
      movie={
        Object {
          "id": 4,
          "release_date": "2019-10-18",
          "title": "Joker",
        }
      }
      poster="https://image.tmdb.org/t/p/w342undefined"
      title="Joker"
      toggleFavorites={[MockFunction]}
    />
    <Card
      favorited={true}
      movie={
        Object {
          "id": 56,
          "release_date": "2019-10-04",
          "title": "Maleficent: Mistress of Evil",
        }
      }
      poster="https://image.tmdb.org/t/p/w342undefined"
      title="Maleficent: Mistress of Evil"
      toggleFavorites={[MockFunction]}
    />
    <Card
      favorited={true}
      movie={
        Object {
          "id": 70,
          "release_date": "2019-10-18",
          "title": "Zombieland: Double Tap",
        }
      }
      poster="https://image.tmdb.org/t/p/w342undefined"
      title="Zombieland: Double Tap"
      toggleFavorites={[MockFunction]}
    />
    <img
      alt="right arrow"
      className="right-arrow arrow"
      src="left-arrow.png"
    />
  </div>
</div>
`;
